File: ActorClient/ActorClient/CocoaCryptoProvider.m
//
//  CocoaCryptoProvider.m
//  ActorClient
//
//  Created by Антон Буков on 27.02.15.
//  Copyright (c) 2015 Anton Bukov. All rights reserved.
//

#import <CommonCrypto/CommonDigest.h>
#import "J2ObjC_source.h"
#import "CocoaCryptoProvider.h"

@implementation CocoaCryptoProvider

- (IOSByteArray *)SHA256WithByteArray:(IOSByteArray *)data
{
    IOSByteArray *ret = [IOSByteArray arrayWithLength:32];
    CC_SHA256(data.buffer, data.length, ret.buffer);
    return ret;
}

@end

File: ActorClient/ActorClient/CocoaCryptoProvider.swift
//
// Created by Stepan Korshakov on 10.03.15.
// Copyright (c) 2015 Anton Bukov. All rights reserved.
//

import Foundation

@objc class CocoaCryptoProvider: NSObject, JavaObject, AMCryptoProvider  {
    func SHA256WithByteArray(data: IOSByteArray) -> IOSByteArray {
        var ret = IOSByteArray(length: 32);
        CC_SHA256(data.buffer(), CC_LONG(data.length()), UnsafeMutablePointer<UInt8>(ret.buffer()));
        return ret;
    }
}
